x-pack/plugins/security_solution/public/timelines/components/timeline/event_details.tsx

```diff
@@ -11,39 +11,105 @@
  */
 
 import { EuiSpacer } from '@elastic/eui';
-import React from 'react';
+import React, { useCallback, useEffect, useMemo } from 'react';
 import deepEqual from 'fast-deep-equal';
+import { find } from 'lodash/fp';
 
+import { useDispatch } from 'react-redux';
 import { BrowserFields, DocValueFields } from '../../../common/containers/source';
 import {
   ExpandableEvent,
   ExpandableEventTitle,
+  OnEventDetailsClose,
 } from '../../../timelines/components/timeline/expandable_event';
 import { useDeepEqualSelector } from '../../../common/hooks/use_selector';
+import { TimelineEventsDetailsItem } from '../../../../common/search_strategy/timeline';
+import { useTimelineEventsDetails } from '../../containers/details';
+import { TimelineId } from '../../../../common/types/timeline';
+import { activeTimeline } from '../../containers/active_timeline_context';
+import { timelineActions } from '../../store/timeline';
 
 interface EventDetailsProps {
   browserFields: BrowserFields;
   docValueFields: DocValueFields[];
   timelineId: string;
+  onEventDetailsClose?: OnEventDetailsClose;
 }
 
 const EventDetailsComponent: React.FC<EventDetailsProps> = ({
   browserFields,
   docValueFields,
   timelineId,
+  onEventDetailsClose,
 }) => {
-  const expandedEvent = useDeepEqualSelector(
-    (state) => state.timeline.timelineById[timelineId]?.expandedEvent
+  const dispatch = useDispatch();
+
+  const { expandedEvent, isSaving } = useDeepEqualSelector(
+    (state) => state.timeline.timelineById[timelineId]
   );
 
+  const [loading, detailsData] = useTimelineEventsDetails({
+    docValueFields,
+    indexName: expandedEvent.indexName!,
+    eventId: expandedEvent.eventId!,
+    skip: !expandedEvent.eventId,
+  });
+
+  const isAlert = useMemo(() => {
+    if (detailsData) {
+      const signalField = find({ category: 'signal', field: 'signal.rule.id' }, detailsData) as
+        | TimelineEventsDetailsItem
+        | undefined;
+
+      if (signalField?.originalValue) {
+        return true;
+      }
+    }
+    return false;
+  }, [detailsData]);
+
+  const handleOnEventClosed = useCallback(() => {
+    dispatch(
+      timelineActions.toggleExpandedEvent({
+        timelineId,
+        event: null,
```

Comment:
maybe we could call it with an empty object and don't have to modify store?
https://github.com/elastic/kibana/pull/83963/files#diff-8b70b403fc95374f549151c97e7ef0d41da31fa4f286cdd6083d9b2157c38264R186